### PR TITLE
Remove obsolete finalizer from KnativeServing resources.

### PR DESCRIPTION
**Pull request by markusthoemmes**
## Proposed Changes

The genreconciler machinery adds a finalizer automatically. There's no need to define our own and do our own management of it. We need to make sure to remove it from old resources though.

**Release Note**

```release-note
NONE
```

/assign @houshengbo @jcrossley3

### CODE REVIEW

File: pkg/reconciler/knativeserving/knativeserving.go [hunk 1]
```diff
@@ -252,24 +255,40 @@ func (r *Reconciler) checkDeployments(ctx context.Context, manifest *mf.Manifest
 }
 
 // ensureFinalizer attaches a "delete manifest" finalizer to the instance
-func (r *Reconciler) ensureFinalizer(ctx context.Context, manifest *mf.Manifest, instance *servingv1alpha1.KnativeServing) error {
-	for _, finalizer := range instance.GetFinalizers() {
-		if finalizer == finalizerName {
-			return nil
-		}
+func (r *Reconciler) ensureFinalizerRemoval(_ context.Context, _ *mf.Manifest, instance *servingv1alpha1.KnativeServing) error {
```
Comment (houshengbo) [inline pkg/reconciler/knativeserving/knativeserving.go:258]: If I understand correctly, the comment of this function should be changed into 
```
ensureFinalizerRemovel ensures that Finalizers in the operator Serving CR is consistent with the ones generated by genreconciler.
```
The `oldFinalizerName = "delete-knative-serving-manifest"` is removed, if it exists in old operator serving CR.

In the `delete` function, we do not even check the finilizers any more.

Comment (markusthoemmes) [reply pkg/reconciler/knativeserving/knativeserving.go:258]: Good catch, changed the comment.

**Pull request merged**